app/controllers/amr_heatmap_controller.rb

```diff
@@ -2,10 +2,10 @@ class AmrHeatmapController < ApplicationController
   before_action :admin_required
 
   def index
-    @sample_ids = params[:sampleIds]
+    @sample_ids = params[:sampleIds].map(&:to_i)
```

Comment:
if params[:sampleIds] could be strings, then you would want to also map to_i at the top of #amr_counts, right?